src/Mvc/Mvc.ApiExplorer/src/ApiResponseTypeProvider.cs

```diff
@@ -77,13 +77,48 @@ private static List<IApiResponseMetadataProvider> GetResponseMetadataAttributes(
            IReadOnlyList<IApiResponseMetadataProvider> responseMetadataAttributes,
            Type? type,
            Type defaultErrorType)
+        {
+            var contentTypes = new MediaTypeCollection();
+
+            var responseTypes = ReadResponseMetadata(responseMetadataAttributes, type, defaultErrorType, contentTypes);
+
+            // Set the default status only when no status has already been set explicitly
+            if (responseTypes.Count == 0 && type != null)
+            {
+                responseTypes.Add(new ApiResponseType
+                {
+                    StatusCode = StatusCodes.Status200OK,
+                    Type = type,
+                });
+            }
+
+            if (contentTypes.Count == 0)
+            {
+                // None of the IApiResponseMetadataProvider specified a content type. This is common for actions that
+                // specify one or more ProducesResponseType but no ProducesAttribute. In this case, formatters will participate in conneg
+                // and respond to the incoming request.
+                // Querying IApiResponseTypeMetadataProvider.GetSupportedContentTypes with "null" should retrieve all supported
+                // content types that each formatter may respond in.
+                contentTypes.Add((string)null!);
+            }
+
+            CalculateResponseFormats(responseTypes, contentTypes);
+
+            return responseTypes;
+        }
+
+        // Shared with EndpointMetadataApiDescriptionProvider
+        internal static List<ApiResponseType> ReadResponseMetadata(
+            IReadOnlyList<IApiResponseMetadataProvider> responseMetadataAttributes,
+            Type? type,
+            Type defaultErrorType,
+            MediaTypeCollection contentTypes)
```

Comment:
@pranavkm I decided to share logic for determining response formats with `ApiResponseTypeProvider`. I don't think I changed the behavior.

Reply:
No issues. This type has enough unit tests so as long as they continue passing, this should be fine.